internal/core/src/common/FThreadPools.h

```diff
@@ -0,0 +1,75 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MILVUS_FTHREADPOOLS_H
+#define MILVUS_FTHREADPOOLS_H
+
+#include <memory>
+#include <map>
+#include <shared_mutex>
+#include <folly/executors/CPUThreadPoolExecutor.h>
+#include <folly/futures/Future.h>
+#include <functional>
+#include "common/Common.h"
+
+namespace milvus {
+
+enum FThreadPoolKey {
+    CHUNKCACHE_POOL = 0,
+};
+
+class FThreadPools {
+ public:
+    static void
+    ResizeThreadPool(FThreadPoolKey key, float ratio);
+
+    template <typename F, typename... Args>
+    static auto
+    Submit(FThreadPoolKey key, F&& func, Args&&... args)
+        -> folly::SemiFuture<decltype(func(std::forward<Args>(args)...))> {
+        auto& executor = GetThreadPool(key);
+        auto bound_func =
+            std::bind(std::forward<F>(func), std::forward<Args>(args)...);
+        return folly::via(&executor, std::move(bound_func)).semi();
+    }
+
+    ~FThreadPools() {
+        ShutDown();
```

Comment:
`FThreadPools` is a tool class which will never be instantiated, so this clean will not get called. 
Also `ShutDown` need to be static to manipulate static field 
Let's:
1. rename it to `FThreadPoolsManager`
2. Make `ShutDown` static
3. Add a auto cleaner
```
private:
struct AutoCleanup {
    ~AutoCleanup() { FThreadPools::ShutDown(); }
};
static AutoCleanup cleanup_;
```

I know we copy this code from `ThreadPools`, same probelm need to be handled